pkg/ccl/crosscluster/logical/lww_kv_processor.go

```diff
@@ -321,15 +357,43 @@ func newKVTableWriter(
 	}, nil
 }
 
+// writeableColumns are 'writable' in the sense that they are stored on disk in the primary index.
+//
+// We assume this is all public non-virtual columns and all virtual columns stored in the primary
+// indexes value or key.
+//
+// NOTE: We don't handle virtual columns that are stored in secondary index keys or values here
+// because we assume tables with such columns were disallowed earlier. This function will need to be
+// updated if that restriction is relaxed.
+func writeableColunms(ctx context.Context, td catalog.TableDescriptor) ([]catalog.Column, error) {
+	// TODO(ssd): Validate with SQL foundations that this is correct.
+	keyColumnIDs := td.GetPrimaryIndex().CollectKeyColumnIDs()
+
+	pubCols := td.PublicColumns()
+	ret := make([]catalog.Column, 0, len(pubCols))
+	for _, c := range pubCols {
+		if c.IsComputed() && c.IsVirtual() && !keyColumnIDs.Contains(c.GetID()) {
```

Comment:
sry, why do we need to check `keyColumnIDs`?

Reply:
Because a virtual column can be included in the primary key, in which case we must provide its value. Now, this is all academic at the moment because it looks like we've disallowed virtual columns in public keys a level above. But, I think a good number of those would actually work since if the virtual column is in the public key we do get it in the rangefeed event unless it is a composite-encoded values.